src/common/templates/_eso-secrets-helper.tpl

```diff
@@ -239,3 +239,141 @@ spec:
     {{- end }}
 {{- end }}
 {{- end }}
+
+{{/*
+Generates Random secrets for the provided secret keys in helpers.tpl or direct usage in secret file
+USAGE:
+{{- include "harnesscommon.generateSecretsInHelpers" (dict "ctx" $) }}
+*/}}
+{{- define "harnesscommon.generateSecretsInHelpers" }}
+{{- $ := .ctx }}
+{{- $hasAtleastOneSecret := false }}
+{{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+{{- range $key, $value := .ctx.Values.secrets.default }}
+    {{- $hasAtleastOneSecret = true }}
+    {{- $pathTosSecret := printf "secrets.default.%s" $key }}
+{{ $key }}: {{ include "harnesscommon.secrets.passwords.manage" (dict "secret" $.Chart.Name "key" $key "providedValues" (list $pathTosSecret ) "length" 10 "context" $) }}
+{{- end }}
+{{- end }}
+
+{{/*
+Renders secrets as environment variables for the provided secret keys in deployment.yaml/sts.yaml
+USAGE:
+{{- include "harnesscommon.renderSecretsAsEnvironmentVariables" (dict "ctx" $) }}
+*/}}
+{{- define "harnesscommon.renderSecretsAsEnvironmentVariables" }}
+{{- $ := .ctx }}
+{{- $hasAtleastOneSecret := false }}
+{{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+{{- $defaultSecretList := .ctx.Values.secrets.default }}
+{{- $kubernetesSecretsList := (index .ctx.Values.secrets.kubernetesSecrets 0).keys }}
+{{- $ESOSecretsList := (index .ctx.Values.secrets.secretManagement.externalSecretsOperator 0).remoteKeys }}
+{{- $fileSecretMap := (index .ctx.Values.secrets.fileSecret 0).keys }}
+{{- $mergedSecretKeys := keys $defaultSecretList $kubernetesSecretsList $ESOSecretsList | uniq }}
+{{- range $key := $mergedSecretKeys }}
+    {{- if not (has $key $fileSecretMap) }}
```

Comment:
there can be a usecase where the filesecret needs an environment variable for the file path as well
For eg. for redis SSL CA cert, there is a file mounted secret and an environment variable which points to the mounted path. Applications use that environment variable to load the file contents

Reply:
that can be passed in config.yml , redis helper functions remain unchanged and will not use this pattern due to easier maintainence from global fields in SMP releases.